### PR TITLE
Remove shards weight file support

### PR DESCRIPTION
**Pull request by howard0su**
ref: #1991

### CODE REVIEW

**Review by ggerganov — APPROVED** [review]
Merge after resolving the conflicts

File: llama.cpp [hunk 1]
```diff
@@ -686,31 +581,17 @@ struct llama_model_loader {
...
     }
 
     bool alignment_prevents_mmap() {
         for (const llama_load_tensor & lt : tensors_map.tensors) {
-            for (const llama_load_tensor_shard & shard : lt.shards) {
-                if (shard.file_off & 3) {
-                    return true;
-                }
+            if (lt.file_off & 3) {
+                return true;
```
Comment (howard0su) [inline llama.cpp:589]: @ggerganov do we still have the problem that a tensor offset is not aligned? I believe it is not.

Comment (slaren) [reply llama.cpp:589]: It shouldn't happen with current models, only with pre-ggjt models.

Comment (howard0su) [reply llama.cpp]: thank you. I will remove this check.

**Comment by howard0su** [comment]
As I changed additional code after review, can you review again? @ggerganov

**Comment by ggerganov** [comment]
ref #1991

**Pull request merged**